compiler/noirc_printable_type/src/lib.rs

```diff
@@ -168,27 +168,35 @@ fn fetch_printable_type(
 fn to_string(value: &PrintableValue, typ: &PrintableType) -> Option<String> {
     let mut output = String::new();
     match (value, typ) {
-        (
-            PrintableValue::Field(f),
-            PrintableType::Field
-            // TODO(#2401): We should print the sign for these and probably print normal integers instead of field strings
-            | PrintableType::SignedInteger { .. }
-            | PrintableType::UnsignedInteger { .. },
-        ) => {
+        (PrintableValue::Field(f), PrintableType::Field) => {
             output.push_str(&format_field_string(*f));
         }
-        (PrintableValue::Field(f), PrintableType::Boolean) => {
-            if f.is_one() {
-                output.push_str("true");
-            } else {
-                output.push_str("false");
+        (PrintableValue::Field(f), PrintableType::UnsignedInteger { width }) => {
+            let uint_cast = f.to_u128() & ((1 << width) - 1); // Retain the lower 'width' bits
+            output.push_str(&uint_cast.to_string());
+        }
+        (PrintableValue::Field(f), PrintableType::SignedInteger { width }) => {
+            let mut uint = f.to_u128(); // Interpret as uint
+
+            // Extract sign relative to width of input
+            if (uint >> (width - 1)) == 1 {
+                output.push('-');
+                uint = (uint ^ ((1 << width) - 1)) + 1; // Two's complement relative to width of input
```

Comment:
Similarly, this could be `uint as i128 * -1`. I think the whole block could be simpler if line 179 were changed to `let mut int = f.to_u128() as i128;` Actually, in that case I think we can skip the sign check entirely and convert the i128 directly to a string.